dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -248,15 +248,16 @@ public static String generateResultSummary(VerificationTask task, ProverEnvironm
                 printWarningIfThreadStartFailed(p, encCtx, prover);
                 if (props.contains(PROGRAM_SPEC) && FALSE.equals(model.evaluate(PROGRAM_SPEC.getSMTVariable(encCtx)))) {
                     summary.append("===== Program specification violation found =====\n");
-                    for(Event e : p.getThreadEvents(Local.class)) {
-                        if(e.hasTag(Tag.ASSERTION) && TRUE.equals(model.evaluate(encCtx.execution(e)))) {
+                    for(Assert ass: p.getThreadEvents(Assert.class)) {
+                        if(FALSE.equals(model.evaluate(encCtx.encodeExpressionAsBooleanAt(ass.getExpression(), ass)))) {
```

Comment:
Why false? is the condition in the new assert event reverted wrt locals?

Reply:
The original code checked for the assertion being executed not for its condition. So the old code implicitly assumed that we only ever have `assert(false)` (I think some pieces of code still have this assumption).